src/librustc_codegen_llvm/intrinsic.rs

```diff
@@ -951,11 +956,19 @@ fn codegen_msvc_try(
             Some(did) => bx.get_static(did),
             None => bug!("eh_catch_typeinfo not defined, but needed for SEH unwinding"),
         };
-        let funclet = catchpad.catch_pad(cs, &[tydesc, bx.const_i32(0), slot]);
+        let flags = bx.const_i32(8); // Catch by reference
+        let funclet = catchpad.catch_pad(cs, &[tydesc, flags, slot]);
 
-        let payload = catchpad.load(slot, i64_align);
+        let i64_align = bx.tcx().data_layout.i64_align.abi;
+        let payload_ptr = catchpad.load(slot, ptr_align);
+        let payload = catchpad.load(payload_ptr, i64_align);
```

Comment:
AFAICT, this is loading from a non-defined value -- unless I'm missing something. The slot itself has type `*mut [i64; 2]` but I don't see any code here that would initialize it (or allocate stack space for the `[i64; 2]` value). Maybe that's done implicitly by `catch_pad`?

Reply:
The slot is initialized by the `catch_pad` instruction. The `[i64; 2]` is located in the stack frame that actually threw the exception, which is still active at this point (inside the catchpad).